src/client.ml

```diff
@@ -125,7 +129,6 @@ let execute_multi_result ?params conn query =
 (* Execute [f iter] for the first result set iterator and throw an exception if there is more than
    one result set *)
 let execute_f' ?params ~f conn query =
-  let f = Scheduler.preserve_execution_context' f |> Staged.unstage in
```

Comment:
why is this not necessary anymore?

Reply:
This was what actually caused the hang in execute_map. It will make logging slightly worse in certain cases but it does something weird with exception handling.